protocol/contracts/nexus/Shard.sol

```diff
@@ -0,0 +1,555 @@
+pragma solidity 0.8.18;
```

Comment:
Currently doesn't compile under hh, `yarn compile`

Temple v2 already pulls in 0.8.19, so perhaps easier to just update these new contracts to 0.8.19 too?